src/components/show-products/index.js

```diff
@@ -1,25 +1,64 @@
-import React from 'react';
+import { doc, setDoc, serverTimestamp } from '@firebase/firestore';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 import { useHistory } from 'react-router';
-import Button from '../button';
-import { updatePurchaseDate } from '../../utils/firebaseUtils';
+import { db } from '../../lib/firebase';
+
+//Hooks
 import { useProducts } from '../../hooks/useProducts';
 import {
   useLocalStorage,
   LOCAL_STORAGE_LIST_TOKEN,
 } from '../../hooks/useLocalStorage';
+
+//Utils
+import { updatePurchaseDate } from '../../utils/firebaseUtils';
 import { TimeFrames } from '../../utils/timeFrames';
-import { compareDates } from '../../utils/compareDates';
+import {
+  diffBetweenTodayAndDate,
+  ONE_DAY,
+} from '../../utils/diffBetweenTodayAndDate';
+// import { daysSinceLastTransaction } from '../../utils/daysSinceLastTransaction';
+import { findProductById } from '../../utils/findProductById';
 import ContentContainer from '../content-container';
+import Button from '../button';
+
 import './styles.css';
 
 export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
   const timeFrames = TimeFrames;
+  const one_day = ONE_DAY;
 
-  const handleOnChange = (productID) => {
+  const handleOnChange = (event, productID) => {
     updatePurchaseDate(productID, storedValue);
+
+    const item = findProductById(products, productID);
```

Comment:
In this specific case, you can re-use the object returned by `updatePurchaseDate` function